src/App.js

```diff
@@ -1,16 +1,59 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [ chatData, setChatData ] = useState(chatMessages);
+
+  const updateChatData = updatedChat => {
+    const updatedChats = chatData.map((chat) => {
+      if (chat.id === updatedChat.id) {
+        return updatedChat;
+      } else {
+        return chat;
+      }
+    });
+
+    setChatData(updatedChats);
+  };
+
+  // count the number of likes to display in header
+  const likedTally = chatData.reduce((tally, chat) => {
+    if (chat.liked) {
+      tally += 1;
+    }
+
+    return tally;
+  }, 0)
+
+  // establish a local sender to pass down to ChatEntry
+  // for use in toggling CSS classes
+  const localName = chatData[0].sender
```

Comment:
Interesting way of handling establishing the "local" sender! More flexible than using hard-coded strings for sure. In practice, you'd likely have more data than just these chats to be able to identify the "local" user.